workflow_creation/utilities_auto_gui.md

```diff
@@ -1,96 +1,81 @@
-
+Parameter Extraction for CloudiFacturing Generic User Input asynchronous Web Application
+===
+
+Web applications used for printing typically rely on end user input. In order to prevent malicious parameter inputs, workflow providers have to specify valid parameter ranges for inputs. A suitable way to allow for parameter configuration are spreadsheet files. For this purpose, this repository provides the _GenericGUIApplication_ GUI web application in _code\_examples/Python/app\_generic\_gui_. It allows for interactive parameter specification. When the application runs, it initially constructs appropriate GUI elements for the specified parameters. These GUI elements are incorporated into a web page, which will be presented to the user. The remainder of this document describes the requirements, configuration and deployment of the application.
+
+1 Requirements
+---
+
+You only require to have docker installed. The docker container installs all required packages automatically and runs the application.
+
+2 Configuration
+---
+In order to configure the web application, three different configuration files are used. The following sections describe each of them.
+
+#### 2.1 __Environment Variable File for Service Deployment Parameters:__ 
+The _env_ file from the directory containing this README file specifies environment variables for the docker container. When the application runs, it reads these environment variables. The _env_ file contains all parameters for service deployment and also specifies the used configuration files for web site construction. The listing below shows a complete _env_ file for service deployment configuration.
+
+```
+# CONTEXT_ROOT defines the deployment location relative to the host.
+# Set this variable such that it fits to the URL under which the VM hosting the
+# app can be reached.
+CONTEXT_ROOT=/gui-app
+
+# Define deployment parameters and config file names
+# Service name
+SERVICE_NAME=GUIApp
+
+# Target name space
+TNS=tns
+
+# Configuration file for layout
+PAGE_CONFIG=GUIWebPageConfig.xml
+
+# XLS sheet to specify GUI elements
+XLS_PARAM_SHEET=GUIAppParameters.xlsx
+```
+
+#### 2.2 Configuration File for Web Page Layout:
+The _app_ directory contains a configuration file for the web site layout parameters such as font size or title size. The listing below shows a complete layout configuration file. 
+
+```XML
+<pageConfiguration>
+    <params>
+        <param name="layout">top2bottom</param> <!-- Specify the order of GUI elements (top2bottom or left2right)-->
+        <param name="orientation">single</param>
+		<param name="margin">20</param>
+		<param name="margin_direction">left</param>
+		<param name="title_size">4</param> <!-- from 1 (large) to 6 (small) -->
+		<param name="heading_size">5</param> <!-- from 1 (large) to 6 (small) -->
+		<param name="font_size">14px</param> <!-- Size for text other than title and headings -->
+    </params>
+</pageConfiguration>
+```
+
+#### 2.3 Excel-Sheet for GUI Elements:

```

Comment:
I suggest to swap sections 2.2 and 2.3, such that the more relevant part (the Excel sheet) comes first.

Reply:
I assume that this is more convenient for the reader as well.